### PR TITLE
Fix #5443/user not exists when violations is deleted

### PR DESCRIPTION
**Pull request by JoseCaste**
# GreenCityUBS PR
Bug when an admin is trying to delete order violation

## Summary Of Changes
Unit testing, controllers, service layer

## Issue Link
https://github.com/ita-social-projects/GreenCity/issues/5443

##Changes
* service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java
* service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java
# PR Checklist Forms

_(to be filled out by PR submitter)_
- [x] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java [hunk 1]
```diff
@@ -82,19 +83,10 @@ void deleteViolationThrowsException() {
         assertThrows(UserNotFoundException.class, () -> violationService.deleteViolation(1L, "abc"));
     }
 
-    @Test
-    void deleteViolationFromOrderResponsesNotFoundWhenNoViolationInOrder() {
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java:86]: Why did you remove this test?
You should mock  `employeeRepository.findByUuid(uuid)` here

File: service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java [hunk 2]
```diff
@@ -82,19 +83,10 @@ void deleteViolationThrowsException() {
...
-        Assertions.assertThrows(NotFoundException.class, () -> violationService.deleteViolation(1L, "abc"));
-        verify(violationRepository, times(1)).findByOrderId(1L);
-    }
-
     @Test
     void deleteViolationFromOrderByOrderId() {
-        User user = ModelUtils.getTestUser();
-        when(userRepository.findUserByUuid("abc")).thenReturn(Optional.of(user));
+        Employee user = ModelUtils.getEmployee();
+        when(employeeRepository.findByUuid("abc")).thenReturn(Optional.of(user));
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java:99]: please add verify for `employeeRepository.findByUuid("abc")`

File: service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java [hunk 3]
```diff
@@ -180,13 +182,13 @@ public void deleteViolation(Long id, String uuid) {
 
     @Override
     public void updateUserViolation(UpdateViolationToUserDto add, MultipartFile[] multipartFiles, String uuid) {
-        User currentUser = userRepository.findUserByUuid(uuid)
+        Employee currentUser = employeeRepository.findByUuid(uuid)
             .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java:186]: please add test for case when  `.orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));`

File: service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java [hunk 4]
```diff
@@ -157,8 +158,9 @@ public Optional<ViolationDetailInfoDto> getViolationDetailsByOrderId(Long orderI
     @Override
     @Transactional
     public void deleteViolation(Long id, String uuid) {
-        User currentUser = userRepository.findUserByUuid(uuid)
+        Employee currentUser = employeeRepository.findByUuid(uuid)
             .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java:162]: please add test for this case

File: service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java (hunk 3 above)
Comment (JoseCaste) [reply service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java:186]: it was already added

File: service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java [hunk 5]
```diff
@@ -84,24 +85,24 @@ void deleteViolationThrowsException() {
 
     @Test
     void deleteViolationFromOrderResponsesNotFoundWhenNoViolationInOrder() {
-        User user = ModelUtils.getTestUser();
-        when(userRepository.findUserByUuid("abc")).thenReturn(Optional.of(user));
+        Employee employee = ModelUtils.getEmployee();
+        when(employeeRepository.findByUuid("abc")).thenReturn(Optional.of(employee));
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/service/ubs/ViolationServiceImplTest.java:89]: please add `verify` for this mock

File: service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java [hunk 6]
```diff
@@ -171,7 +173,7 @@ public void deleteViolation(Long id, String uuid) {
             User user = violationOptional.get().getOrder().getUser();
             user.setViolations(userRepository.countTotalUsersViolations(user.getId()));
             userRepository.save(user);
-            eventService.saveEvent(OrderHistory.DELETE_VIOLATION, currentUser.getRecipientEmail(),
+            eventService.save(OrderHistory.DELETE_VIOLATION, currentUser.getEmail(),
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/service/ubs/ViolationServiceImpl.java:176]: Why did you change `saveEvent` to `save`?

**Review by RomaMocherniuk — APPROVED** [review]

**Review by juseti — APPROVED** [review]

**Review by JJJazl — APPROVED** [review]

**Review by MaksymGolik — APPROVED** [review]
